src/laboratory/api/shelfobject.py

```diff
@@ -332,6 +335,11 @@ def delete(self, request, org_pk, lab_pk, **kwargs):
         :return:
         """
         self._check_permission_on_laboratory(request, org_pk, lab_pk, "delete")
+        serializer = ShelfObjectDeleteSerializer(data=request.data, laboratory=self.laboratory)
+        serializer.is_valid(raise_exception=True)
+        utils.organilab_logentry(self.request.user, serializer.validated_data['shelfobj'], DELETION, relobj=self.laboratory)
+        serializer.validated_data['shelfobj'].delete()
```

Comment:
This is not actually deleting things from database, since what you get from the serializer is the PK and not the object, so if you check the database after the "delete was successful" the row is still there untouched (this is an important test you cannot forget to make).

You can get the shelf_object this way: `shelf_object = get_object_or_404(ShelfObject.objects.filter(in_where_laboratory=lab_pk), pk=serializer.data['shelf_object'])`

Reply:
Ok, this was my mistake, the code actually receives a shelf object and deletes it correctly, thing is I made a mistake testing it (changed something shouldn't have).  I apologize, you don't need to change anything here.